src/modules/translations/services/translations.service.ts

```diff
@@ -0,0 +1,31 @@
+import { CACHE_MANAGER, Inject, Injectable, Logger } from '@nestjs/common';
+import { Cache } from 'cache-manager';
+import { get } from 'lodash';
+
+import { Translations } from '../types';
+
+import { GET_SITE_VARIABLES_BY_NAME } from './queries.gql';
+
+import { DataService } from '~modules/data/services/data.service';
+@Injectable()
+export class TranslationsService {
+	private logger: Logger = new Logger(TranslationsService.name, { timestamp: true });
+
+	constructor(
+		private dataService: DataService,
+		@Inject(CACHE_MANAGER) private cacheManager: Cache
+	) {}
+
+	public async getTranslations(): Promise<Translations> {
+		let translations: Translations = await this.cacheManager.get('translations');
+		if (!translations) {
+			const response = await this.dataService.execute(GET_SITE_VARIABLES_BY_NAME, {
+				name: `translations-frontend`,
+			});
+			translations = get(response, 'data.cms_site_variables[0]');
```

Comment:
lets throw a 404 not found if the value is undefined with message: "No translations have been set in the database"

Reply:
Wouldn't we want to return only the values?
```suggestion
			translations = get(response, 'data.cms_site_variables[0].value');
```

Reply:
updated